packages/validator/source/validator.ts

```diff
@@ -118,7 +123,17 @@ export class Validator implements Contracts.Validator.Validator {
 		const candidateTransactions: Contracts.Crypto.Transaction[] = [];
 		const failedTransactions: Contracts.Crypto.Transaction[] = [];
 
+		// txCollatorFactor% of the time for block preparation, the rest is for  block and proposal serialization and signing
+		const timeLimit =
+			performance.now() +
+			this.cryptoConfiguration.getMilestone().timeouts.blockPrepareTime *
```

Comment:
I think this doesn't work as intended when `blockPrepareTime` is 0 or very small as is currently the case for our node running the tx pool api tests: https://github.com/ArkEcosystem/mainsail/blob/feat/valdiator/time-limit-block-creation/tests/functional/transaction-pool-api/paths/config/crypto.json#L3432 because the loop always exits immediately.

One solution could be to enforce a minimum `timeLimit` / `blockPrepareTime`.

Reply:
This is likely causing https://github.com/ArkEcosystem/mainsail/actions/runs/9549641061/job/26344431896?pr=620 to never finish

Reply:
blockPrepareTime is increased